docs/kb/use-functions.md

```diff
@@ -24,12 +24,63 @@ The steps below describe a generalized deployment step of a function on the VMwa
 
 ## Knative
 
-For this walk-through, the [`kn-ps-echo`](/examples) function from the example folder is used.
+For most this walk-through, the [`kn-ps-echo`](/examples) function from the examples folder is used. For working with secrets, the [`kn-ps-email`](/examples) function from the examples folder is used.
 
 ### Knative Prerequisites
 
 Before proceeding to deploy a function, you must have VMware Event Broker Appliance deployed and be able to SSH to the appliance or have access to the kubernetes configuration file (/root/.kube/config) locally on your desktop to authenticate
 
+### Knative Function Deployment with Secrets
+
+Most functions require configuration parameters to work correctly. A function author can hard code values inside the function. However, any changes to the configuration parameters means the function's container image must be rebuilt. By using configuration parameters stored in a Kubernetes secret, a function can be reconfigured by simply changing the secret.
+
+You can verify whether a function requires a secret by looking at the function's `function.yaml`. Below is part of the `function.yaml` for `kn-ps-email`. Note the `secretRef:` section specifying the name of the secret the function expects.
```

Comment:
```suggestion
You can verify whether a function requires a secret by looking at the function's `function.yaml`. The `kn-ps-echo` function does not use secrets because the function simply echoes any event that it receives.

Below is part of the `function.yaml` for `kn-ps-email`. Note the `secretRef:` section specifying the name of the secret the function expects.
```

Reply:
Moved